nomad/structs/structs.go

```diff
@@ -7872,6 +7881,22 @@ func (t *Task) Validate(jobType string, tg *TaskGroup) error {
 		}
 	}
 
+	// Validate actions.
+	actions := make(map[string]bool)
+	for _, action := range t.Actions {
+		if err := action.Validate(); err != nil {
+			outer := fmt.Errorf("Action %s validation failed: %s", action.Name, err)
+			mErr.Errors = append(mErr.Errors, outer)
+		}
+
+		if handled, seen := actions[action.Name]; seen && !handled {
+			mErr.Errors = append(mErr.Errors, fmt.Errorf("Action %s defined multiple times", action.Name))
+			actions[action.Name] = true
+			continue
+		}
```

Comment:
This way to check for duplicates is a bit different from others because it handles 3+ repetitions as a single error. Not sure if that's better, but if it's we could apply the same logic to other blocks.

Reply:
:+1: I like this approach. We don't include file context from the original HCL, so I feel like having repeats of the error doesn't get us any new information.